tf2onnx/onnx_opset/reduction.py

```diff
@@ -144,6 +144,29 @@ def version_6(cls, ctx, node, **kwargs):
         ctx.make_node(op_type="Greater", inputs=[reduce_node_output, zero_node.output[0]],
                       name=node.name, outputs=node.output, shapes=shapes, dtypes=dtypes)
 
+    @classmethod
+    def version_13(cls, ctx, node, **kwargs):
+        keepdims = node.get_attr_value('keep_dims')
+        reduce_input = node.input[0]
+        if node.type == "All":
+            reduce_input = ctx.make_node("Not", [reduce_input]).output[0]
+        cast = ctx.make_node("Cast", inputs=[reduce_input], attr={"to": onnx_pb.TensorProto.FLOAT}).output[0]
```

Comment:
I was wondering why it is necessary to cast into float?

Reply:
Hmm, good question.  I copied over the existing code and just modified it to do dynamic axes.  I think one of the previous ops maybe didn't support non-float values.  In any case, float is technically good because it doesn't roll over and is guaranteed to preserve monotonicity of addition.  Since we just want to detect if any of the values are non-zero, this is good.